services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/share_activity_pack/shareToStudents.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react'
+
+import ActivityPackInformation from './activityPackInformation';
+import ShareActivityPackModal from './shareActivityPackModal';
+
+import AssignmentFlowNavigation from '../../assignment_flow_navigation'
+import ScrollToTop from '../../../shared/scroll_to_top'
+import { Card, ExpandableCard } from '../../../../../Shared/index'
+import {
+  UNIT_TEMPLATE_NAME,
+  UNIT_TEMPLATE_ID,
+  UNIT_NAME,
+  ACTIVITY_IDS_ARRAY,
+  CLASSROOMS,
+  UNIT_ID,
+  ASSIGNED_CLASSROOMS
+} from '../../assignmentFlowConstants'
+import { Activity, ActivityPack } from '../../../../../../interfaces/activityPack';
+
+const addStudentsSrc = `${process.env.CDN_URL}/images/illustrations/add-students.svg`
+const addShareActivityPackSrc = `${process.env.CDN_URL}/images/icons/icons-share-activity-pack.svg`
+const shareActivitySrc = `${process.env.CDN_URL}/images/icons/icons-share.svg`
+
+export const ShareToStudents = ({ activityPackData, moveToStage4 }) => {
+
+  const unitId = window.localStorage.getItem(UNIT_ID);
+  const classrooms = JSON.parse(window.localStorage.getItem(ASSIGNED_CLASSROOMS));
+
+  const [leaving, setLeaving] = React.useState<boolean>(false);
+  const [isExpanded, setIsExpanded] = React.useState<boolean>(false);
+  const [modalOpen, setModalOpen] = React.useState<boolean>(false);
+  const [activityPack, setActivityPack] = React.useState<ActivityPack>(activityPackData);
+  const [singleActivity, setSingleActivity] = React.useState<Activity>(null);
+
+  React.useEffect(() => {
+    setActivityPack(activityPackData);
+  }, [activityPackData]);
+
+  React.useEffect(() => {
+    if (leaving) {
+      handleUnmount()
+    }
+  }, [leaving]);
```

Comment:
Why do we need this when the effect below should already handle the cleanup?

Reply:
yea sorry there's a slight difference in the second PR. There's a redirect to the classrooms section of the dashboard that we don't want to happen when the user leaves this stage to move to the `Stage4` component